views/js/runner/plugins/security/watermark.js

```diff
@@ -0,0 +1,441 @@
+        style: '',
+        /**
+         * Override default watermark container styles
+         * @type {String}
+         * @example `padding: 50px`
+         */
+        containerStyle: '',
+        /**
+         * Long hash string will be visually split into the shorter text parts with N chars each.
+         * (sha-1 is 40 chars, so 4 parts by 10 chars. sha-256 is 64 chars, so 6 parts by 10 chars and ignore lefotver 4)
+         *  @type {Number}
+         */
+        textPartLength: 10,
+        /**
+         * Visually, these text parts will be separated by sequence of N spaces.
+         * Recommended: `15` if 'foreground', `3` if 'background', `1` if 'circle.
+         *  @type {Number}
+         */
+        separatorsLength: 3,
+
+        /**
+         * Enable/disable watermark in the runtime.
+         * Click on free space in the item -> type `triggerWord` to open dialog ->
+         *   in the dialog type keyword which matches configured `hash` & `algorithm`.
+         * NB! letters in `triggerWord` conflict with test-runner shortcuts.
+         *   For example, `triggerWord` is 'abr'.
+         *   But typing 'r' toggles review panel.
+         *   Then can either use `ABR` - uppercase letters typed with 'Shift',
+         *   or keep 'abr' but type it while holding 'Alt'.
+         */
+        unlock: {
+            enabled: false,
+            triggerWord: 'WMK',
+            algorithm: 'SHA-512',
+            hash: ''
+        },
+
+        /**
+         * Profiles to use if any of the settings above (except `unlock`)
+         *   should depend on the specific test/item
+         * @example
+         * a) add category `x-tao-option-watermark-abc` for one test, and `x-tao-option-watermark-def` for another test
+         * b) set `{
+         *    configsByCategory : {
+         *      abc: {type: 'circle', style: 'color:blue'},
+         *      def: {type: 'foreground', style: 'color:red'}
+         *   }`
+         * c) first test will show blue circle, second test will show red foreground
+         */
+        configsByCategory: {}
+    };
+
+    const watermarkTypes = {
+        foreground: 'foreground',
+        background: 'background',
+        circle: 'circle'
+    };
+
+    /**
+     * TODO: move to `@oat-sa\tao-core-sdk\src\util\shortcut\registry.js`,
+     *   to have all existing test-runner shortucts in the one place,
+     *   and to be able to configure desired modifier keys (Alt+Shift...)
+     * But at the moment shortcut registry doesn't support shortcuts with multiple letter keys
+     * @param {string} triggerWord - letter key sequence; does distinguish uppercase and lowercase ("R" is different from "r")
+     * @param {Function} action
+     * @param {Object} abortController - to remove listener
+     * @param {HTMLElement[]} freeSpaceTargets - checks key events with target on body and these elements
+     */
+    function addMultikeyShortcut(triggerWord, action, abortController, freeSpaceTargets) {
```

Comment:
It's not a big issue, but the secret key sequence also brings up the dialog on items which don't have the watermark category - you could avoid it with an early return based on `isPluginEnabled()` or `is('hidden')`.